### PR TITLE
Add ability to search contacts by sales tag

### LINKED ISSUE #116: As an organised salesman, I can find contacts who have purchased items of a certain sales tag
...so that I can gain insights in the type of items my contacts are interested in.

### PR DESCRIPTION
**Pull request by Asthenosphere**
Closes #116 

Screenshot 2020-10-18 at 3 57 49 PM

### CODE REVIEW

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -240,28 +240,60 @@ public void editSaleTag(Tag target, Tag editedTag) {
...
-    public int findByContactTag(Tag target) {
-        return (int) persons.asUnmodifiableObservableList()
-                .stream()
-                .filter(p -> p.getTags().contains(target)).count();
+    public String findByContactTag(Tag target) {
+        StringBuilder output = new StringBuilder();
+        int count = 0;
+        for (Person p : persons) {
+            if (p.getTags().contains(target)) {
+                output.append(String.format("%d. %s\n", count++ + 1, p.toString()));
```
Comment (sebastiantoh) [inline src/main/java/seedu/address/model/AddressBook.java]: Does using `++count` instead work? Same for below

**Review by sebastiantoh — COMMENTED** [review]
Nice, works well! 

Maybe you might want to update the user guide as well? Also, I'm guessing the `c/` flag only works when searching for a sales tag and has no effect when searching for a contact tag. Maybe can include this in the user guide as well (that including `c/` flag when searching for a contact tag has no effect) so that people don't think it's a bug or something.

File: src/main/java/seedu/address/model/AddressBook.java (hunk 1 above)
Comment (Asthenosphere) [reply src/main/java/seedu/address/model/AddressBook.java]: Thanks!

**Review by sebastiantoh — APPROVED** [review]
LGTM!

**Pull request merged**